addons/crm/models/crm_team.py

```diff
@@ -505,6 +505,11 @@ def _allocate_leads(self, work_days=1):
             # assign + deduplicate and concatenate results in teams_data to keep some history
             candidate_lead = teams_data[team]["leads"][0]
             assign_res = team._allocate_leads_deduplicate(candidate_lead, duplicates_cache=duplicates_lead_cache)
+            if 'leads_with_different_company' in assign_res:
```

Comment:
Still not convinced by this. 

Can we not make this part into its own method and override it instead? Instead of checking a key that won't ever be there without the sales_crm addon?

Reply:
@maan-odoo Hey, just checking that you saw this :)

Reply:
Yes, I saw it!
The same issue (multiple companies) is also found in CRM, so I am trying to solve it with this PR. It may take time.

Reply:
This is fine now as it's used in CRM too and the check generally makes sense